src/WebJobs.Extensions.DurableTask.Analyzers/Analyzers/ActivityFunction/FunctionAnalyzer.cs

```diff
@@ -128,27 +142,92 @@ private bool TryGetFunctionNameFromActivityInvocation(InvocationExpressionSyntax
             return false;
         }
 
-        private bool TryGetInputNodeFromCallActivityInvocation(InvocationExpressionSyntax invocationExpression, out SyntaxNode inputNode)
+        private static bool TryGetInputNodeFromCallActivityInvocation(SemanticModel semanticModel, InvocationExpressionSyntax invocationExpression, out SyntaxNode inputNode)
+        {
+            // If CallActivity is a custom method in user code
+            if (SyntaxNodeUtils.TryGetDeclaredSyntaxNode(semanticModel, invocationExpression, out SyntaxNode declaration))
+            {
+                if (TryGetSpecificParameterIndex(declaration, "object input", out int inputParameterIndex))
+                {
+                    if (TryGetInvocationArguments(invocationExpression, out IEnumerable<ArgumentSyntax> arguments))
+                    {
+                        var argumentNode = arguments.ElementAt(inputParameterIndex);
+                        inputNode = argumentNode.ChildNodes().First();
+                        return true;
+                    }
+                }
+            }
+            // else assume CallActivity is a DurableFunctions method
```

Comment:
Only by checking if the arguments match the parameters expected. But yes. I avoided this because it was more complicated than checking if the method was defined in user code, but I realize now that this is the better option.

I can use reflection to get the methods and parameters I'm interested in looking at and make sure that the arguments on the CallActivity* call match one of our methods.

This can also be used to make sure I'm looking at the correct input parameter to prevent a manual change to the analyzer in case the input parameter isn't the last parameter in some new overload that we could possibly add in the future (Connor's suggestion above).